tests/model_tests/test_interfaces.py

```diff
@@ -286,6 +286,29 @@ def test_device_hostname(device):
                                          name='eth0')
 
 
+def test_interface_networks_refresh(device):
+    """Test the interface parent networks refresh upon reparenting of a
+    Network object"""
+    cidr = '10.1.1.1/32'
+    parent_network = models.Network.objects.create(
+        cidr='10.1.1.0/24', site=device.site
+    )
+    intf_address = models.Network.objects.create(
+        cidr=cidr, site=device.site
+    )
+    intf = models.Interface.objects.create(device=device, name='eth0')
+    intf.assign_address(cidr)
+    intf.clean_addresses()
+    assert intf.get_networks() == ['10.1.1.0/24']
+
+    new_parent_network = models.Network.objects.create(
+        cidr='10.1.1.0/27', site=device.site
+    )
+
+    intf_obj = models.Interface.objects.get(device=device, name='eth0')
+    intf_obj.clean_addresses()
```

Comment:
This new `refresh_assignment_interface_networks()` signal calls `clean_addresses()`, right? That seems to be what we're trying to test, so doesn't calling `int_obj.clean_addresses()` here defeat the purpose of the test?

Reply:
You're sorta right, however, if you look at L300 where I assign the interface a network /32, I have to call `.clean_addresses()` on the interface object explicitly for the network assignment to show up in the call to `get_networks()` ..... This is a pattern I noticed in some other tests within the code as well.

Reply:
The one on line 300 makes sense, since `assign_addresses()` doesn't perform a `save()` on the interface. I think your need to call `clean_addresses()` here is because you don't call `save()` around L301, since `clean_addresses()` on L301 doesn't save the Interface to the database. 

Since the Interface is never getting saved to the database, your Network creation on L304 has nothing to update (so your bug fix doesn't get exercised), and I bet that's why you're having to call `clean_addresses()` here to get the test to pass.

Reply:
That makes total sense! I guess I didn't think of the `.save()` at all! I've pushed up the changes to the test now :)